extensions/ql-vscode/src/query-history.ts

```diff
@@ -492,8 +492,9 @@ export class QueryHistoryManager extends DisposableObject {
     singleItem: QueryHistoryInfo,
     multiSelect: QueryHistoryInfo[]
   ): Promise<void> {
+    // TODO will support remote queries
     const { finalSingleItem, finalMultiSelect } = this.determineSelection(singleItem, multiSelect);
-    if (!this.assertSingleQuery(finalMultiSelect) || finalSingleItem.t !== 'local') {
+    if (!this.assertSingleQuery(finalMultiSelect) || (finalSingleItem && finalSingleItem.t !== 'local')) {
```

Comment:
Any reason you're not doing `finalSingleItem?.t !== 'local'` like in other places?

Reply:
Yes. In this case, if the selected query is `remote`, then we want this command to do nothing. If there is no selection, then we want to throw an error. It is trying to preserve previous behaviour.

However, in a future PR, I got rid of the throw clause. It's actually really difficult for a user to get into a situation where this command is invoked, but no query is selected. Also, if it does happen, then popping up an error message is a bit annoying.

So, for this PR, I was just planning on maintaining previous behaviour.